### ISSUE TITLE
I want to redirect into my application when user accept call. not to native calling screen

### ISSUE THREAD

**Issue by saurabh874** [react-native-webrtc/react-native-callkeep#308]
Hi I am using react-native v-0.61.4 with react-native-callkeep.
is there anyay to redirect into own application when user accept call. not to native calling screen?

**Comment by samad324** [comment]
yes, use ```RNCallKeep.backToForeground();``` method. this method will bring your app on the top of the native call screen.

**Comment by saurabh874** [comment]
@samad324 thanks but in iOS. it always open native call kit. when app is in back-ground or kill state.
do you have any idea how to resolve it and will it be possible to remove native call screen when accept a call?

**Comment by samad324** [comment]
add `answerCall` listener in your app and redirect on your call screen. it will automatically direct you on your app.

```
 RNCallKeep.addEventListener('answerCall', async ({callUUID}) => {
     this.props.navigation.navigate('Video');
});
```

**Comment by saurabh874** [comment]
but it is not working when app is in kill mode or phone locked.

**Comment by samad324** [comment]
ohh.. my app is working fine when app is killed, but yes i'm facing the issue when phone is locked. and trying to resolve it as well. Kindly comment here if you will find any solution.

**Comment by saurabh874** [comment]
Sure @samad324

**Comment by samad324** [comment]
@saurabh874 hi, i saw the behaviour of the WhatsApp and Skype on lock screen, and they behaved same. if you want see your app you need to unlock your device.

**Comment by agamsps** [comment]
Any update @samad324 I am still facing this issue.
I receive a push on locked ios, slide to answer, enter the passcode and the app does not opens.
TIA

**Comment by samad324** [comment]
Sorry, @agamsps , I got switched to another project and I don't know how they handle this.

**Comment by namnm** [comment]
When phone is locked, there's no way to open app directly from PN.
https://stackoverflow.com/questions/41740139

There will be an icon in the native dialer, if user click on that icon, the phone will show unlock screen and if user unlock the phone, app will be shown. There's a config `imageName` when you setup CallKeep, which map to the objective C `iconTemplateImageData` to customize the image of that icon.

**Comment by gabrielelana** [comment]
@agamsps see my solution in last comment of #355

**Comment by namnm** [comment]
@gabrielelana Great info. Thanks for sharing

**Comment by Doma-byte** [comment]
If app is in foreground and in background then navigation works perfectly, but the issue reside when app has been killed by user. Using fcm i trigger the callkeep UI with setbackgroundhandler but after accepting the call it's just navigate to native callUI screen not the app. Did anyone solve this issue?

**Comment by GabrielBermudez** [comment]
> If app is in foreground and in background then navigation works perfectly, but the issue reside when app has been killed by user. Using fcm i trigger the callkeep UI with setbackgroundhandler but after accepting the call it's just navigate to native callUI screen not the app. Did anyone solve this issue?

were you able to solve it?

**Comment by namnm** [comment]
> > If app is in foreground and in background then navigation works perfectly, but the issue reside when app has been killed by user. Using fcm i trigger the callkeep UI with setbackgroundhandler but after accepting the call it's just navigate to native callUI screen not the app. Did anyone solve this issue?
> 
> were you able to solve it?

On ios, it is not possible to invoke and show app UI when screen locked or app killed with voip callkit.
You need to use the native call UI screen. Then there is a button there to open your app if users want:
https://stackoverflow.com/questions/41740139

**Comment by Irfanwani** [comment]
On iOS, I don't think we can redirect users to app directly if the device is locked (though when device is unlocked, and if we accept the call via the call notification, the default behaviour is to open the app as user interacted with the notification, thus you can easily navigate to wherever you want).

In Kill state, it is better to handle the call via the native call screen only. You can add all the listeners (E.g, DTMF etc) to listen to the actions made via native call listener or you can also disable some of these actions (E.g, disable video call option if not offered by your app) by setting the appropriate options (E.g, supportsVideo: false in iOS options in react native callkeep setup method).

Other than this, the native call screen shows one more button in the bottom left corner (the icon if this button can be configured via `imageName` option in ios object in setup options. You need to mention the name of the image which should be added to the project [in ios directory]).

If that button is clicked, that opens the app which is handling the call.
